src/components/datetime/DateTimeController.cpp

```diff
@@ -117,4 +117,30 @@ char const* DateTime::MonthsString[] = {"--", "JAN", "FEB", "MAR", "APR", "MAY",
 char const* DateTime::MonthsStringLow[] = {"--", "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
 
 char const* DateTime::MonthsLow[] = {
-  "--", "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"};
\ No newline at end of file
+  "--", "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"};
+
+void DateTime::GetTimeStr(char* string, bool is24h) {
+  char minutesChar[3];
+  sprintf(minutesChar, "%02d", static_cast<int>(minute));
+  char hoursChar[3];
+  char ampmChar[3] = "  ";
+  if (is24h) {
+    sprintf(hoursChar, "%02d", hour);
+  } else {
+    if (hour == 0) {
+      hour = 12;
+      sprintf(ampmChar, "AM");
+    } else if (hour < 12) {
+      sprintf(ampmChar, "AM");
+    } else if (hour >= 12) {
+      hour = hour - 12;
+      sprintf(ampmChar, "PM");
+    }
+    sprintf(hoursChar, "%02d", hour);
+    if (hoursChar[0] == '0') {
+      hoursChar[0] = ' ';
+    }
+  }
+  
+  sprintf(string, "%c%c:%c%c %c%c", hoursChar[0], hoursChar[1], minutesChar[0], minutesChar[1], ampmChar[0], ampmChar[1]);
```

Comment:
This can be reduced down to:
```
if (is24h) {
  sprintf(string, "%02u:%02u   ", hour, minute);
} else {
  char const* const ampmStr = hour < 12 ? "AM" : "PM;
  
  auto tempHour = hour;
  if (tempHour == 0) {
    tempHour = 12;
  } else if (tempHour > 12) {
    tempHour -= 12;
  }

  sprintf(string, "%2u:%02u %s", tempHour, minute, ampmStr);
}
```